### PR TITLE
Solve issues with sherpa 4.17 and pin version for doc build

### PR DESCRIPTION
**Pull request by registerrier**
**Description**

This pull request solves issues caused by the sherpa v4.17, solving an issue with the sherpa backend that must have appeared with the numpy v2.0. For some reason, the all dependencies environment in the CI never installed sherpa (at least since we moved to numpy>=2) and silently skipped the test on sherpa. 

- The default `np.nan` for parameter min and maxes are causing issues with sherpa. They are now changed to `-np.inf` and `np.inf`.
- sherpa >4.16 is imposed in environment-dev.yml to make sure we use numpy>=2
- a restriction in the docs requirement is introduced in setup.cfg (sherpa<4.16). This is to keep a working environment with numpy <2 . We are still stuck to astropy<=5.3 for docs build. This should be solved but it will require an independent PR.

**Dear reviewer**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -51,7 +51,7 @@ console_scripts =
 [options.extras_require]
 all =
     naima
-    sherpa; platform_system != "Windows"
+    sherpa<4.17; platform_system != "Windows"
```
Comment (bkhelifi) [inline setup.cfg:54]:
```suggestion
    sherpa<4.16; platform_system != "Windows"
```

Comment (bkhelifi) [reply setup.cfg:54]: if the title of your PR is correct...

**Comment by bkhelifi** [comment]
For information, on Ubuntu, I have launch the command `conda install -c https://cxc.cfa.harvard.edu/conda/sherpa sherpa=4.15`.
It finds the library, which is an improvement.

BUT, it forces to downgrade numpy to 1.26!!!
```
The following packages will be DOWNGRADED:

  algopy                                 0.7.2-pyhd8ed1ab_0 --> 0.6.1-pyhd8ed1ab_0 
  numpy                               2.1.1-py311h71ddf71_0 --> 1.26.4-py311h64a7726_0 

```

And this is in conflict with our environement that requires `numpy>1.20`.

It think that this PR does not solve yet #5233

**Comment by bkhelifi** [comment]
the same issue with sherpa v4.16. It also requires numpy 1.26!

File: environment-dev.yml [hunk 2]
```diff
@@ -77,7 +77,7 @@ dependencies:
   - pyinstrument
   - memray
   - pip:
-      - sherpa
+      - sherpa>=4.17
```
Comment (MRegeard) [inline environment-dev.yml:80]:
```suggestion
      - sherpa<=4.17
```

File: setup.cfg [hunk 3]
```diff
@@ -51,15 +51,15 @@ console_scripts =
 [options.extras_require]
 all =
     naima
-    sherpa; platform_system != "Windows"
+    sherpa>=4.17; platform_system != "Windows"
```
Comment (MRegeard) [inline setup.cfg:54]:
```suggestion
    sherpa<=4.17; platform_system != "Windows"
```

File: setup.cfg [hunk 4]
```diff
@@ -51,15 +51,15 @@ console_scripts =
...
     healpy; platform_system != "Windows"
     requests
     tqdm
     ipywidgets
     ray[default]>=2.9
 all_no_ray =
     naima
-    sherpa; platform_system != "Windows"
+    sherpa>=4.17; platform_system != "Windows"
```
Comment (MRegeard) [inline setup.cfg:62]:
```suggestion
    sherpa<=4.17; platform_system != "Windows"
```

**Review by MRegeard — CHANGES_REQUESTED** [review]
Thanks @registerrier.
Putting `sherpa<=4.17` here does the job for me.
You can also put `sherpa<4.17` in the docs section.
This way you don't have conflicts and it will install 4.17 (numpy2) for the tests and 4.16 (numpy<2) for the docs.

**Review by MRegeard — APPROVED** [review]
@registerrier Thank you very much !

File: environment-dev.yml (hunk 2 above)
Comment (bkhelifi) [reply environment-dev.yml:80]: I am not sure about that... The dev should handle new versions! 
The same type of modifications in the other subsections have been merged, but I am not sure about it.

File: setup.cfg [hunk 5]
```diff
@@ -80,6 +80,7 @@ test =
 docs =
     astropy>=5.0,<5.3
     numpy<2.0
+    sherpa<4.16
```
Comment (bkhelifi) [inline setup.cfg:83]: See my comment above

**Review by bkhelifi — CHANGES_REQUESTED** [review]
Thanks a lot @registerrier. You are solving a long-standing issue! I have just a remaining test

File: setup.cfg (hunk 5 above)
Comment (registerrier) [reply setup.cfg:83]: Thanks @bkhelifi . I have updated the motivation of the PR to explain why.

**Review by bkhelifi — APPROVED** [review]

**Pull request merged**